bond: Add support of balance-slb option

cathay4t opened the pull request:
The NetworkManager 1.41 introduced new bond option `balance-slb` to achieve
the same effect with ovs bond slb mode.

Example yaml:

```yml
---
interfaces:
- name: bond99
  type: bond
  state: up
  link-aggregation:
    mode: balance-xor
    options:
      xmit_hash_policy: "vlan+srcmac"
      balance-slb: 1
```

Unit and integration test cases included.

ffmancera reviewed (approved):
LGTM! Let's wait CI

ffmancera reviewed (changes requested):
The integration tests need to be marked as xfail/skip for nm-stable runs. In addition, there is a tier2 bond test failing as the bond is not found. Could this be related with selinux? As NetworkManager is running a script for nft if selinux is enabled it could be block it and then Nmstate is not finding the bond interface. Please, take a look thanks!

cathay4t committed:
bond: Add support of balance-slb option

The NetworkManager 1.41 introduced new bond option `balance-slb` to
achieve the same effect with ovs bond slb mode.

Example yaml:

```yml
---
interfaces:
- name: bond99
  type: bond
  state: up
  link-aggregation:
    mode: balance-xor
    options:
      xmit_hash_policy: "vlan+srcmac"
      balance-slb: 1
```

Unit and integration test cases included.

Integration test case is shipped in CI as bond slb require patched
kernel which is not in github CI ubuntu host yet.

Signed-off-by: Gris Ge <fge@redhat.com>

cathay4t commented:
It is not SElinux issue. The CI host kernel does not support new nft command. I have marked the test case to skip in CI environment.

cathay4t commented:
Manually tested on CentOS stream 9 with NetworkManager-1.41.4-31323.copr.9e2ff9dabd.el9.x86_64

ffmancera reviewed (approved):
LGTM! Tested manually as well
